app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailFragment.kt

```diff
@@ -0,0 +1,723 @@
+                    }
+                    EventDetailViewModel.ShowListOfAttendeesButtonTypes.NOT_GOING -> {
+                        goingTextView.isVisible = false
+                        goingAttendeeRecyclerView.isVisible = false
+                        notGoingTextView.isVisible = true
+                        notGoingAttendeeRecyclerView.isVisible = true
+                        allButtonTextView.setTextColor(
+                            ResourcesCompat
+                                .getColor(resources, R.color.dark_gray, null)
+                        )
+                        allButton.background.setTint(
+                            ResourcesCompat
+                                .getColor(resources, R.color.light_2, null)
+                        )
+                        goingButtonTextView.setTextColor(
+                            ResourcesCompat
+                                .getColor(resources, R.color.dark_gray, null)
+                        )
+                        goingButton.background.setTint(
+                            ResourcesCompat
+                                .getColor(resources, R.color.light_2, null)
+                        )
+                        notGoingButtonTextView.setTextColor(
+                            ResourcesCompat
+                                .getColor(resources, R.color.white, null)
+                        )
+                        notGoingButton.background.setTint(
+                            ResourcesCompat
+                                .getColor(resources, R.color.black, null)
+                        )
+                    }
+                }
+            }
+            allButton.setOnClickListener {
+                viewModel.showAllAttendees()
+            }
+            goingButton.setOnClickListener {
+                viewModel.showGoingAttendees()
+            }
+            notGoingButton.setOnClickListener {
+                viewModel.showNotGoingAttendees()
+            }
+        }
+        collectLatestLifecycleFlow(viewModel.goingAttendees) { goingAttendees ->
+            val goingAttendeeAdapter = AttendeeItemAdapter(goingAttendees,
+                isAttendee,
+                onDeleteIconClick = {attendee ->  viewModel.deleteAttendee(attendee)})
+            binding.goingAttendeeRecyclerView.adapter = goingAttendeeAdapter
+            binding.goingAttendeeRecyclerView.layoutManager = LinearLayoutManager(requireContext())
+        }
+        collectLatestLifecycleFlow(viewModel.notGoingAttendees){ notGoingAttendees ->
+            val notGoingAttendeeAdapter = AttendeeItemAdapter(
+                notGoingAttendees,
+                isAttendee,
+                onDeleteIconClick = {attendee ->  viewModel.deleteAttendee(attendee)})
+            binding.notGoingAttendeeRecyclerView.adapter = notGoingAttendeeAdapter
+            binding.notGoingAttendeeRecyclerView.layoutManager = LinearLayoutManager(requireContext())
+        }
+    }
```

Comment:
I wouldn't recreate the `LinearLayoutManager` on every emission here. You just need to setup your RecyclerView once, but then you don't need to assign adapter and layout manager again. Just toggle its visibility and you're good.
So all that should happen here is that you update the attendees list of your adapter. Same counts for the `goingAttendees` collector above.

Reply:
So I should just move "binding.notGoingAttendeeRecyclerView.layoutManager = LinearLayoutManager(requireContext())" outside the scope of the flow? And the same as goingAttendees

Reply:
Yes, the only thing you need to update is the list the RV holds